packages/client/builtin/KaTexBlockWrapper.vue

```diff
@@ -20,53 +20,76 @@ Learn more: https://sli.dev/guide/syntax.html#latex-line-highlighting
 -->
 
 <script setup lang="ts">
-import { range, remove } from '@antfu/utils'
 import { computed, getCurrentInstance, inject, onMounted, onUnmounted, ref, watchEffect } from 'vue'
+import type { PropType } from 'vue'
 import { parseRangeString } from '@slidev/parser'
-import { CLASS_VCLICK_TARGET, injectionClicks, injectionClicksDisabled, injectionClicksElements } from '../constants'
-import { makeId } from '../logic/utils'
+import { CLASS_VCLICK_HIDDEN, CLASS_VCLICK_TARGET, injectionClicks } from '../constants'
+import { makeId, safeParseNumber } from '../logic/utils'
 
 const props = defineProps({
   ranges: {
+    type: Array as PropType<string[]>,
     default: () => [],
   },
+  finally: {
+    type: [String, Number],
+    default: 'last',
+  },
   startLine: {
     type: Number,
     default: 1,
   },
   at: {
-    type: Number,
-    default: undefined,
+    type: [String, Number],
+    default: 'flow',
   },
 })
 
-const clicks = inject(injectionClicks)
-const elements = inject(injectionClicksElements)
-const disabled = inject(injectionClicksDisabled)
+const clicks = inject(injectionClicks)?.value
 
 const el = ref<HTMLDivElement>()
 const vm = getCurrentInstance()
 
 onMounted(() => {
-  const prev = props.at == null ? elements?.value.length : props.at
+  if (!clicks || clicks.disabled)
+    return
+
+  const at = props.at === 'flow' ? '+1' : props.at
+  const atNum = safeParseNumber(at)
+  const inFlow = typeof at === 'string' && '+-'.includes(at[0])
+  const flowSize = inFlow
+    ? atNum + props.ranges.length - 2
+    : 0
+  const start = inFlow
+    ? clicks.flowSum + atNum - 1
+    : atNum
+  const end = start + props.ranges.length - 1
```

Comment:
This chunk of code as been duplicate in a few components and it mighjt be hard to sync the changes later one - maybe we could have `parseAtProp` retrns an object with extra info.